lib/normalizer.c

```diff
@@ -3949,6 +3949,12 @@ grn_nfkc_normalize_remove_target_non_blank_character_p(
   if (data->options->char_type_func(current) == GRN_CHAR_SYMBOL) {
     return data->options->remove_symbol;
   }
+  /* `unify_middle_dot` is `true` and the unified middle dot is handled as a
+   * symbol. */
+  if (data->options->unify_middle_dot &&
+      grn_nfkc_normalize_is_middle_dot_family(current, current_length)) {
+    return data->options->remove_symbol;
+  }
   return false;
```

Comment:
ノーマライズしてから`remove_symbol`チェックをする方がキレイなんだけど、それはしんどいってことだよね？

できるだけ余計な処理をしたくないから、先に`remove_symbol`チェックをしちゃおうか。

```suggestion
  if (!data->options->remove_symbol) {
    return false;
  }

  if (data->options->char_type_func(current) == GRN_CHAR_SYMBOL) {
    return true;
  }
  /* `unify_middle_dot` is `true` and the unified middle dot is handled as a
   * symbol. */
  if (data->options->unify_middle_dot &&
      grn_nfkc_normalize_is_middle_dot_family(current, current_length)) {
    return true;
  }

  return false;
```

あと、`remove_symbol`のときは`unify_middle_dot`のノーマライズはしないようにしてくれる？

Reply:
そうですね、順番を変えればいいじゃないか！と思って眺めてたんですが、影響範囲が広そうだったのでこの方針にしました。

Reply:
一応、本当はノーマライズしてからがいいんだけど、こういう実装にしちゃっているっていうコメントを残しておいて。